packages/graphql/src/translate/translate-read.ts

```diff
@@ -118,28 +143,38 @@ export function translateRead({
         const edgeVar = new Cypher.NamedVariable("edge");
         const totalCountVar = new Cypher.NamedVariable("totalCount");
 
-        const withCollect = new Cypher.With([Cypher.collect(nodeVarRef), edgesVar]).with(edgesVar, [
+        const withCollect = new Cypher.With([Cypher.collect(matchNode), edgesVar]).with(edgesVar, [
             Cypher.size(edgesVar),
             totalCountVar,
         ]);
 
-        const unwind = new Cypher.Unwind([edgesVar, nodeVarRef]).with(nodeVarRef, totalCountVar);
+        const unwind = new Cypher.Unwind([edgesVar, matchNode]).with(matchNode, totalCountVar);
         connectionPreClauses = Cypher.concat(withCollect, unwind);
 
         const connectionEdge = new Cypher.Map({
             node: projectionExpression,
         });
 
-        const withTotalCount = new Cypher.With([connectionEdge, edgeVar], totalCountVar, nodeVarRef);
+        const withTotalCount = new Cypher.With([connectionEdge, edgeVar], totalCountVar, matchNode);
         const connectionClause = getConnectionSortClause({ context, projection, varName });
         const returnClause = new Cypher.With([Cypher.collect(edgeVar), edgesVar], totalCountVar).return([
             new Cypher.Map({
                 edges: edgesVar,
                 totalCount: totalCountVar,
             }),
-            nodeVarRef,
+            matchNode,
         ]);
 
+        if (!projectionSubqueries.empty && hasOrdering) {
+            addSortAndLimitOptionsToClause({
+                optionsInput,
+                target: matchNode,
+                projectionClause: returnClause,
+                node,
+                context,
+            });
+        }
+
```

Comment:
I see you've been making changes to our order translation.

We should run the performance tests before emerging this PR to make sure that it has not been affected (we can check at that together)